### ISSUE TITLE
try_err false positive / incorrect suggestion with Poll return type

### ISSUE THREAD

**Issue by tmiasko** [rust-lang/rust-clippy#5855]
The `Poll` type has two implementations of `Try` trait, for `Poll<Result<T, E>>` and `Poll<Option<Result<T, E>>>`.

Currently try_err lint does't take those into account and generates incorrect fix suggestion, for example in code below:

```rust
use std::task::Poll;
use std::io;

pub fn poll(n: usize) -> Poll<Option<io::Result<()>>> {
    match n {
        0 => Err(io::ErrorKind::NotFound)?,
        1 => return Err(io::ErrorKind::NotFound)?,
    // Incorrect suggestion for both of above:
    //    => return Err(io::ErrorKind::NotFound.into()),
    // Possible replacement:
        3 => return Poll::Ready(Some(Err(io::ErrorKind::NotFound.into()))),
        _ => {},
    }
    Poll::Ready(None)
}
```

Additionally, given overall verbosity of replacement code, it is far from clear that replacing `?` is an improvement in this case.
